### PR TITLE
Row-level TTL PR 2: TtlResolver and RemoteTableSpec cleanup

### PR DESCRIPTION
**Pull request by ableegoldman**
Introduces the TtlResolver used to hook everything up within Responsive. Also cleans up the RemoteTableSpec

PR 1: API https://github.com/responsivedev/responsive-pub/pull/370

Future TODO (in order)
1. Cassandra Fact tables (insert/get)
2. TopologyTestDriver
3. Mongo KV tables (insert/get/range/all)
4. Cassandra KVTables (insert/get/range/all)

### CODE REVIEW

File: kafka-client/src/main/java/dev/responsive/kafka/internal/stores/TtlResolver.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    this.ttlProvider = ttlProvider;
+  }
+
+  public TtlDuration defaultTtl() {
+    return ttlProvider.defaultTtl();
+  }
+
+  public boolean hasConstantTtl() {
+    return ttlProvider.hasDefaultOnly();
+  }
```
Comment (agavra) [inline kafka-client/src/main/java/dev/responsive/kafka/internal/stores/TtlResolver.java:62]: 😉

File: kafka-client/src/main/java/dev/responsive/kafka/internal/stores/TtlResolver.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+  public boolean needsValueToComputeTtl() {
+    return ttlProvider.needsValueToComputeTtl();
+  }
+
+  public Optional<TtlDuration> computeTtl(final Bytes keyBytes, final byte[] valueBytes) {
```
Comment (agavra) [inline kafka-client/src/main/java/dev/responsive/kafka/internal/stores/TtlResolver.java:68]: can we add a quick comment on why we need both of these APIs?

**Review by agavra — APPROVED** [review]
LGTM! This one is a quick one :) love more red lines than green ones

File: kafka-client/src/main/java/dev/responsive/kafka/internal/stores/TtlResolver.java (hunk 1 above)
Comment (ableegoldman) [reply kafka-client/src/main/java/dev/responsive/kafka/internal/stores/TtlResolver.java:62]: ahh whoops

**Pull request merged**